docs/development.rst

```diff
@@ -18,3 +18,21 @@ Then setup the Git pre-commit hook for `Black <https://github.com/psf/black>`__
 .. code-block:: console
 
     pre-commit install
+
+Publishing
+----------
+
+Publishing to `PyPI <https://pypi.org/project/pyhf/>`__ and `TestPyPI <https://test.pypi.org/project/pyhf/>`__
+is automated through the `PyPA's PyPI publish GitHub Action <https://github.com/pypa/gh-action-pypi-publish>`__.
+To publish a release to PyPI one simply needs to run
+
+.. code-block:: console
+
+    bumpversion [major|minor|patch]
+
+to update the release version and get a tagged commit and then push the commit
+and tag to :code:`master` with
+
+.. code-block:: console
+
+    git push origin master --tags
```

Comment:
@kratsg this way you both publish `master` branch and **all** tags which may be problematic.

You could push the specific tag rather than pushing all, I guess.
```suggestion
    git push origin master <specific_tag_here>
```

Moreover, you could adopt some wrapper like tox/nox/invoke to wrap a number of commands and pre-validation into a script that would also protect you from creating tags in unwanted places.

The point here is that this whole process should be triggered by a single "button click" rather than have any manual actions in between because humans are error-prone...

Reply:
I would love for a way to have three buttons to click for major/minor/patch and it automatically bumpversions and pushes the tag for us.

Reply:
I actually have a DEMO of such a button in the octomachinery repo. But the button implementation requires a GitHub App.
Basically, my app publishes a "Deploy" page via Checks API on each commit on master. That page contains a clickable button (it's a native feature of Checks on GitHub) and that click triggers a "deployment" event.
So then there's a Workflow that's triggered by the `deployment` event and does the actual work.
You can add up to 3 buttons to a single Checks page. So you could integrate it with calls to `bumpversion` with 3 different options :)
The source is here: https://github.com/sanitizers/diactoros-github-app.
atm it's not configurable but could be tweaked to be :)

Reply:
This is nice! How do you handle permissions with this (so that not _anyone_ can trigger?)

Reply:
The button for the last commit in octomachinery is here: https://github.com/sanitizers/octomachinery/runs/289377465. You don't see it but I do. The last time I checked it only showed up for users with `write` permission on the repo. I filed an FR to make it more flexible but haven't heard back. But if you want to make ACL even stricter, the App would need to be modified to check this. The button click is an event and is delivered to the App using webhooks just like any other events.
The current button click processing is fairly simple: https://github.com/sanitizers/diactoros-github-app/blob/6e68364/diactoros/__main__.py#L60-L86. It just verifies that the button has `deploy` ID and punches GitHub API to create a deployment request, also specifying that it doesn't matter if any of the other CI checks are green (by default, API'd return an error if at the time of the API request there'd be any non-green checks posted on that commit).

Reply:
You can join me complaining about the ACL here: https://github.community/t5/GitHub-API-Development-and/FR-Make-possible-to-show-action-buttons-posted-via-Checks-API-to/m-p/21423/highlight/true#M1342